dash/development/base_component.py

```diff
@@ -280,6 +287,55 @@ def traverse_with_paths(self):
                     for p, t in i.traverse_with_paths():
                         yield "\n".join([list_path, p]), t
 
+    def validate(self):
+        # Make sure arguments have valid values
+        DashValidator.set_component_class(Component)
+        validator = DashValidator(
+            self._schema,
+            allow_unknown=True,
+        )
+        args = {
+            k: self.__dict__[k]
+            for k in self.__dict__['_prop_names']
+            if k in self.__dict__.keys()
+        }
```

Comment:
I don't like this, shouldn't need to get from `__dict__`, `getattr` would be cleaner I think.


```python
args = {
        k: v
        for k, v in ((x, getattr(self, x, None)) for x in self._prop_names)
        if v
}
```

Reply:
Yeah that looks better, not sure what I was thinking there..